dartagnan/src/main/java/com/dat3m/dartagnan/program/expression/type/NumberType.java

```diff
@@ -0,0 +1,5 @@
+package com.dat3m.dartagnan.program.expression.type;
+
+public final class NumberType implements Type {
+    NumberType() {}
+}
```

Comment:
`NumberType` is a bad name because it is highly unspecific (literally everything can be called a number).
If the separation of types is important, then I would rather use `IntegerType` and `BitVectorType` for mathematical vs. programmatical integers. This is close to logic terminology.
If one wants to stay closer to the programmatic side, i.e., use `IntegerType` for bounded integers, then we should rename `NumberType` to something like `MathIntegerType`.
I kinda dislike the usage of "Integer" in both names, but it is still better than `NumberType`.